Issue #5830 Remove com.sun.nio.file import.

janbartel opened the pull request:
Closes #5830

The bnd tool which generates the osgi manifest for jetty-util is including an `Import-Package` for `com.sun.nio.file`, even though this class is only speculatively loaded (ie it is not mandatory for it to be present).

By using the 3-arg Class.forName(String name, boolean initialize, ClassLoader loader) method call, we can get bnd to ignore this package.

BTW: the speculative loading of the com.sun.nio.file.SensitivityWatchEventModifier will probably be removed in jetty-10 onwards, as will any references in jetty-util module-info.java.

janbartel committed:
Issue #5830 Remove com.sun.nio.file import.

This is acheived by using the 3-arg Class.forName() method call.
The bnd tool which generates the osgi manifests doesn't seem to
include Import-Package statements for the 3-arg method call, but
does for the 1-arg method call. The speculative loading of the
com.sun.nio.file.SensitivityWatchEventModifier will probably be
removed in jetty-10 onwards.

Signed-off-by: Jan Bartel <janb@webtide.com>

joakime reviewed (approved):
:+1: - this is a simple enough fix.

olamy commented:
note it's a `nitpicky comment` :)
 what will we happen if in the future bnd tools do not ignore anymore the 3-arg method.
It would be better to use the bdn option `-noclassforname: true` https://bnd.bndtools.org/instructions/noclassforname.html
But not sure how to pass the option...

janbartel commented:
@olamy I thought of that, but decided not to use it because a) every time I've put in special bnd voodoo it comes back later to bite us on the ass in unexpected ways  b) if sometime in the future we added a class that actually wanted the class.forname discovered then by then we would probably have forgotten all about the special flag we added and waste a lot of time diagnosing the problem C) the class.forname in question is going to be deleted on jetty 10 anyway so no point polluting the bnd args for it.

janbartel committed:
Issue #5830 Remove native PathWatcher code.

Signed-off-by: Jan Bartel <janb@webtide.com>

janbartel commented:
@joakime after our offline conversation, I've removed the code that was doing the reflection to use the native classes: can you re-review to make sure I've removed it correctly?

Also, is there any reason to keep the @Disabled annotation on the PathWatcherTest now the native code has gone??

joakime reviewed (approved):
LGTM
